ci/cscs.yml

```diff
@@ -0,0 +1,88 @@
+include:
+  - remote: "https://gitlab.com/cscs-ci/recipes/-/raw/master/templates/v2/.ci-ext.yml"
+
+stages:
+  - baseimage
+  - build
+  - test
+
+.build_baseimage:
+  stage: baseimage
+  # we create a tag that depends on the SHA value of ci/base.Dockerfile, this way
+  # a new base image is only built when the SHA of this file changes
+  # If there are more dependency files that should change the tag-name of the base container
+  # image, they can be added too.
+  # Since the base image name is runtime dependent, we need to carry the value of it to
+  # the following jobs via a dotenv file.
+  before_script:
+  # include build arguments in hash since we use a parameterized Docker file
+  - DOCKER_TAG=`echo "$(cat $DOCKERFILE) $DOCKER_BUILD_ARGS" | sha256sum | head -c 16`
+  - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/public/$ARCH/base/gridtools-ci:$DOCKER_TAG
+  - echo "BASE_IMAGE=$PERSIST_IMAGE_NAME" >> build.env
+  artifacts:
+    reports:
+      dotenv: build.env
+  variables:
+    DOCKERFILE: ci/base.Dockerfile
+    # change to 'always' if you want to rebuild, even if target tag exists already (if-not-exists is the default, i.e. we could also skip the variable)
+    CSCS_REBUILD_POLICY: if-not-exists
+    DOCKER_BUILD_ARGS: '["CUDA_VERSION=$CUDA_VERSION", "UBUNTU_VERSION=$UBUNTU_VERSION"]'
+# build_baseimage_x86_64:
+#   extends: [.container-builder-cscs-zen2, .build_baseimage]
+#   variables:
+#     CUDA_VERSION: 12.6.2
+#     UBUNTU_VERSION: 22.04  
```

Comment:
For later cleanup?

Reply:
Copied over from gt4py. I'll delete it now. Easy to re-introduce if we want to test on x86 systems later...